columnphysics/icepack_shortwave.F90

```diff
@@ -295,29 +287,24 @@ subroutine shortwave_ccsm3 (aicen,    vicen,    &
                                fswsfc=fswsfc(n),     &
                                fswint=fswint(n),     &
                                fswthru=fswthru(n),   &
-                               fswthru_vdr=l_fswthru_vdr(n),&
-                               fswthru_vdf=l_fswthru_vdf(n),&
-                               fswthru_idr=l_fswthru_idr(n),&
-                               fswthru_idf=l_fswthru_idf(n),&
+                               fswthru_vdr=l_fswthru_vdr,&
+                               fswthru_vdf=l_fswthru_vdf,&
+                               fswthru_idr=l_fswthru_idr,&
+                               fswthru_idf=l_fswthru_idf,&
```

Comment:
OK, here we are in the call to `absorbed_solar` (not enough context is shown in this diff), and in that subroutine `fswthru_vdr` and friends are not optional, so here we have to keep the local counterparts. 

It makes sense to makes these scalars instead, nice cleanup.

Reply:
Should `fswthru_vdr` and friends be optional in `absorbed_solar`?

Reply:
absorbed_solar is a local routine, it's not public.  Also, it operates on 1 category while the fswthru_vdr optional arguments are arrays of size ncat.  So we cannot actually pass the full optional argument thru.  So this means we have to create a local copy for that scalar and copy in/out of the optional arguments.  We could do various things to change this like having absorbed solar operate on all categories at once.  But for now, I just left the basic implementation as it was and adapted.  Those fswthru_vdr are always passed as currently implemented so adding optional wouldn't have an impact.  What we'd need to do in that case is add a new flag to say whether to compute those or not based upon the arguments passed into shortwave_ccsm3.  That won't affect results or performance.  If absorbed_solar becomes public, then we may have to rethink this.